src/encoder.js

```diff
@@ -388,62 +419,79 @@ export default class Encoder extends Extendable {
       target.push(tags.BINARY_);
       this.encodeValue(value.byteLength, target);
       this.pushArrayElements(new Uint8Array(value), target);
-    } else if (Array.isArray(value)) {
-      let numElements = value.length;
+    } else {
+      if (this.remainingDepth !== null) {
+        if (this.remainingDepth === 0) {
+          target.push(tags.EXTENSION);
+          if (this.depthBoundExtensionPoint == null) {
+            throw new Error('if remainingDepth != null, depthBoundExtensionPoint must not be either');
```

Comment:
The message should at least match the test being done. Any reason we're using `=== null` in the test?

Reply:
Because I typed this as `number | null`.